src/components/Main/Main.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import {CSSTransition, TransitionGroup} from "react-transition-group";
+import {Route, Switch, useLocation} from "react-router-dom";
+import MainPage from "../pages/MainPage/MainPage";
+import CardsList from "../pages/CardsList/CardsList";
+
+const Main = () => {
+    const location = useLocation();
+    return (
+        <TransitionGroup>
+            <CSSTransition timeout={300} key={location.key} classNames="item">
+                <Switch>
+                    <Route exact path="/main">
+                        <MainPage/>
```

Comment:
It's up to you, but I suppose it's more convenient to use this style of routing:
![Screenshot 2021-09-08 at 11 54 10](https://user-images.githubusercontent.com/50772522/132478817-1e83e361-7a63-42d7-8f1a-945917d0b41a.png)